### PR TITLE
Bananium Essence Reagent

### PR DESCRIPTION
**Pull request by EgorDinamit**
## About The Pull Request

:)

Adds a new funny combat drug. Just... Don't overdose on it.

## Why It's Good For The Game

[Everyone must be happy.](https://www.youtube.com/watch?v=F_LN07MaEPo)

### CODE REVIEW

**Comment by JohnFulpWillard** [comment]
oh god

time to syringe gun people addiction-level amounts.
for a prank, of course.

**Comment by EgorDinamit** [comment]
Might as well make it rarer/harder to produce.
In my opinion, it should take at least a full sheet of bananium for a single unit.

File: ModularTegustation/tegu_chemistry/recipes.dm [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/datum/chemical_reaction/bananium_essence
+	results = list(/datum/reagent/bananium_essence = 1)
+	required_reagents = list(/datum/reagent/bananium = 2, /datum/reagent/drug/happiness = 3, \
+	/datum/reagent/medicine/omnizine = 2, /datum/reagent/medicine/c2/synthflesh = 3) //10u of all this shit = 1u of essence.
+	mix_message = "The mixture laughs at you mockingly..."
```
Comment (retlaw34) [inline ModularTegustation/tegu_chemistry/recipes.dm:5]:
```suggestion
	mix_message = "The mixture laughs menacingly."
```

File: ModularTegustation/tegu_virology.dm [hunk 2]
```diff
@@ -122,3 +122,62 @@
...
+	cure_text = "Bananium Essence."
+	cures = list(/datum/reagent/bananium_essence) // Good luck
+	cure_chance = 1 // As I said - GOOD LUCK.
+	stage_prob = 5
+	agent = "Pure joy"
+	desc = "Nobody can withstand the true power of bananium."
+	severity = DISEASE_SEVERITY_BIOHAZARD
+	visibility_flags = NONE
+	stage1 = list("Your skin itches.")
+	stage2 = list("You catch yourself smiling for no reason.")
```
Comment (retlaw34) [inline ModularTegustation/tegu_virology.dm:137]:
```suggestion
	stage2 = list("You smile.")
```

File: ModularTegustation/tegu_virology.dm [hunk 3]
```diff
@@ -122,3 +122,62 @@
...
+	stage3 = list("<span class='danger'>You feel as if your arm attempts to fall off.</span>", "<span class='danger'>Your skin twitches.</span>", "<span class='danger'>You crave for bananium!</span>")
```
Comment (retlaw34) [inline ModularTegustation/tegu_virology.dm:138]:
```suggestion
	stage3 = list("<span class='danger'>You feel like your arm is  going to fall off!</span>", "<span class='danger'>Your skin twitches.</span>", "<span class='danger'>You could really go for bananium right now!</span>")
```

File: ModularTegustation/tegu_virology.dm [hunk 4]
```diff
@@ -122,3 +122,62 @@
...
+	stage4 = list("<span class='danger'>You hear circus sounds everywhere.</span>", "<span class='danger'>Your feel as if your head exploded!</span>")
```
Comment (retlaw34) [inline ModularTegustation/tegu_virology.dm:139]:
```suggestion
	stage4 = list("<span class='danger'>You hear circus sounds everywhere.</span>", "<span class='danger'>Your feel like your head is about to explode!</span>")
```

File: ModularTegustation/tegu_virology.dm [hunk 5]
```diff
@@ -122,3 +122,62 @@
...
+				affected_mob.say(pick("My skin...", "Urgh...", "Is my head gone..?"))
+			if (prob(5))
+				to_chat(affected_mob, "<span class='danger'>You feel a stabbing pain in your head.</span>")
+				affected_mob.Unconscious(40)
+		if(4)
+			if (prob(50))
+				to_chat(affected_mob, "<span class='danger'>The pain is unbearable!</span>")
+				affected_mob.emote("cry")
+			if (prob(30))
+				to_chat(affected_mob, "<span class='danger'>Your skin begins to move on its own!</span>")
```
Comment (retlaw34) [inline ModularTegustation/tegu_virology.dm:178]:
```suggestion
				to_chat(affected_mob, "<span class='danger'>Your skin begins to shift, hurting like hell!</span>")
```

File: ModularTegustation/tegu_virology.dm [hunk 6]
```diff
@@ -122,3 +122,62 @@
...
+			if (prob(30))
+				affected_mob.emote("laugh")
+			if (prob(20))
+				affected_mob.reagents.add_reagent_list(list(/datum/reagent/drug/happiness = 2))
+		if(3)
+			if (prob(40))
+				affected_mob.emote("smile")
+			if (prob(10))
+				affected_mob.reagents.add_reagent_list(list(/datum/reagent/drug/happiness = 3))
+				affected_mob.say(pick("My skin...", "Urgh...", "Is my head gone..?"))
```
Comment (retlaw34) [inline ModularTegustation/tegu_virology.dm:169]:
```suggestion
				affected_mob.say(pick("Heh... heh", "Honk"))
```

File: ModularTegustation/tegu_virology.dm [hunk 7]
```diff
@@ -122,3 +122,62 @@
...
+			if (prob(50))
+				to_chat(affected_mob, "<span class='danger'>The pain is unbearable!</span>")
+				affected_mob.emote("cry")
+			if (prob(30))
+				to_chat(affected_mob, "<span class='danger'>Your skin begins to move on its own!</span>")
+				affected_mob.emote("scream")
+				affected_mob.Jitter(4)
+			if (prob(10))
+				affected_mob.reagents.add_reagent_list(list(/datum/reagent/drug/happiness = 4))
+				affected_mob.say(pick("Kkkwah...", "Hhhonn...", "K-iilll, me-e...", "The pa-ain..."))
```
Comment (retlaw34) [inline ModularTegustation/tegu_virology.dm:183]:
```suggestion
				affected_mob.say(pick("HO-NK HO-N-N-NK!!", "H-H-ON-NK-K!!", "K-KIL-LL ME HON-NK H-HONK!!", "TH-HE PA-AI-IN!!"))
```

File: ModularTegustation/tegu_virology.dm [hunk 8]
```diff
@@ -122,3 +122,62 @@
...
+	stage_prob = 5
+	agent = "Pure joy"
+	desc = "Nobody can withstand the true power of bananium."
+	severity = DISEASE_SEVERITY_BIOHAZARD
+	visibility_flags = NONE
+	stage1 = list("Your skin itches.")
+	stage2 = list("You catch yourself smiling for no reason.")
+	stage3 = list("<span class='danger'>You feel as if your arm attempts to fall off.</span>", "<span class='danger'>Your skin twitches.</span>", "<span class='danger'>You crave for bananium!</span>")
+	stage4 = list("<span class='danger'>You hear circus sounds everywhere.</span>", "<span class='danger'>Your feel as if your head exploded!</span>")
+	stage5 = list("<span class='danger'>H... O... N... K...</span>")
```
Comment (retlaw34) [inline ModularTegustation/tegu_virology.dm:140]:
```suggestion
	stage5 = list("<span class='danger'>You can only hear honking.</span>")
```

File: ModularTegustation/tegu_chemistry/reagents.dm [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+	reagent_state = SOLID
+	color = "#FFFF00"
+	taste_description = "entire circus"
+	material = /datum/material/bananium
+
+/datum/reagent/bananium_essence
+	name = "Concentrated Bananium Essence"
+	description = "An essence made of pure bananium and some other powerful reagents."
+	color = "#FAFA5A"
+	taste_description = "pure bananium"
```
[the remark this answers is not in the document]
Comment (retlaw34) [reply ModularTegustation/tegu_chemistry/reagents.dm:13]: these should be swapped i feel

**Pull request merged**